File: backend/app/models/File.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.models import Base

class File(Base):
    __tablename__ = 'files'

    file_id = Column(Integer, primary_key=True, autoincrement=True)
    file_name = Column(String(255), nullable=False)
    uploaded_by = Column(Integer, ForeignKey('users.user_id'), nullable=False)
    created_at = Column(DateTime, default=func.now(), nullable=False)
    
    uploaded_by_user = relationship("User", back_populates="uploaded_files")

File: backend/TagFlowApi/Models/File.cs
// TagFlowApi/Models/File.cs
namespace TagFlowApi.Models
{
    public class File
    {
        public int FileId { get; set; }
        public string FileName { get; set; } = "";
        public DateTime CreatedAt { get; set; }

        // Foreign key to User table (who uploaded the file)
        public int UploadedBy { get; set; }
        public User UploadedByUser { get; set; } = null!;

        // One-to-many relationship with file rows
        public ICollection<FileRow> FileRows { get; set; } = new List<FileRow>();
    }
}
